code/modules/atmospherics/machinery/components/machines/atmos_machines_recipes.dm

```diff
@@ -0,0 +1,161 @@
+GLOBAL_LIST_INIT(gas_recipe_meta, gas_recipes_list())
+#define META_RECIPE_ID 1
+#define META_RECIPE_MACHINE_TYPE 2
+#define META_RECIPE_NAME 3
+#define META_RECIPE_MIN_TEMP 4
+#define META_RECIPE_MAX_TEMP 5
+#define META_RECIPE_REACTION_TYPE 6
+#define META_RECIPE_ENERGY_RELEASE 7
+#define META_RECIPE_REQUIREMENTS 8
+#define META_RECIPE_PRODUCTS 9
+
+/proc/gas_recipes_list()
+	. = subtypesof(/datum/gas_recipe)
+	for(var/recipe_path in .)
+		var/list/recipe_info = new(9)
+		var/datum/gas_recipe/recipe = new recipe_path()
+
+		recipe_info[META_RECIPE_ID] = initial(recipe.id)
+		recipe_info[META_RECIPE_MACHINE_TYPE] = initial(recipe.machine_type)
+		recipe_info[META_RECIPE_NAME] = initial(recipe.name)
+		recipe_info[META_RECIPE_MIN_TEMP] = initial(recipe.min_temp)
+		recipe_info[META_RECIPE_MAX_TEMP] = initial(recipe.max_temp)
+		recipe_info[META_RECIPE_REACTION_TYPE] = initial(recipe.reaction_type)
+		recipe_info[META_RECIPE_ENERGY_RELEASE] = initial(recipe.energy_release)
+		recipe_info[META_RECIPE_REQUIREMENTS] = recipe.requirements
+		recipe_info[META_RECIPE_PRODUCTS] = recipe.products
+
+		.[recipe_path] = recipe_info
+
+/proc/recipe_id2path(id)
+	var/list/meta_recipe = GLOB.gas_recipe_meta
+	if(id in meta_recipe)
+		return id
+	for(var/path in meta_recipe)
+		if(meta_recipe[path][META_RECIPE_ID] == id)
+			return path
+	return ""
+
+/datum/gas_recipe
+	var/id = ""
+	var/machine_type = ""
+	var/name = ""
+	var/min_temp = TCMB
+	var/max_temp = INFINITY
+	var/reaction_type = ""
+	var/energy_release = 0
+	var/list/requirements = new/list()
+	var/list/products = new/list()
```

Comment:
```suggestion
	var/list/requirements = list()
	var/list/products = list()
```
Let's not add unnecessary boilerplate.

Reply:
the lists are always full, don't know if a lazy list would be good to have, and i wouldn't know how to handle those in this thing